File: src/CodeAgent.Gateway/Models/AuthModels.cs
namespace CodeAgent.Gateway.Models;

public record RegisterRequest(
    string Email,
    string Password,
    string FirstName,
    string LastName
);

public record LoginRequest(
    string Email,
    string Password,
    bool RememberMe = false
);

public record RefreshTokenRequest(
    string RefreshToken
);

public record AuthResponse
{
    public required string Token { get; init; }
    public required string RefreshToken { get; init; }
    public int ExpiresIn { get; init; }
    public UserDto? User { get; init; }
}

public record UserDto
{
    public required string Id { get; init; }
    public required string Email { get; init; }
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public string[]? Roles { get; init; }
}

File: src/CodeAgent.Web/src/app/core/models/auth.models.ts
export interface User {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  displayName: string;
  avatar?: string;
  roles: string[];
  permissions: string[];
  createdAt: Date;
  lastLogin: Date;
  emailVerified: boolean;
  metadata?: Record<string, any>;
}

export interface AuthTokens {
  accessToken: string;
  refreshToken?: string;
  expiresIn: number;
  tokenType: string;
}

export interface LoginCredentials {
  email: string;
  password: string;
  rememberMe?: boolean;
}

export interface RegistrationData {
  email: string;
  password: string;
  firstName: string;
  lastName: string;
  acceptTerms: boolean;
}

export interface AuthResponse {
  user: User;
  tokens: AuthTokens;
  sessionId: string;
}

export interface AuthState {
  isAuthenticated: boolean;
  user: User | null;
  tokens: AuthTokens | null;
  loading: boolean;
  error: string | null;
}